docs/dev/go-style-guide.md

```diff
@@ -0,0 +1,74 @@
+
+### Go style guide
```

Comment:
I took this from Tendermint and trimmed it down a bit. Feel free to comment on things that should and should not be here.